dashboard/serializers.py

```diff
@@ -39,39 +38,35 @@ def serialize_enrollments(cls, mmtrack, enrollments):
         return serialized_enrollments
 
     @classmethod
-    def serialize_final_grades(cls, mmtrack, enrollments):
+    def serialize_final_grades(cls, mmtrack):
         """
         Serializes a user's final grades
 
         Args:
             mmtrack (MMTrack): An MMTrack object
-            enrollments (iterable): An iterable of CachedEnrollments
         Returns:
             list: Serialized final grades
         """
-        final_grades = []
-        for enrollment in enrollments:
-            try:
-                final_grade = mmtrack.get_final_grade(enrollment.course_run.edx_course_key)
-            except FinalGrade.DoesNotExist:
-                continue
-            if final_grade:
-                final_grades.append({'title': enrollment.course_run.course.title, 'grade': final_grade})
-        return final_grades
+        serialized_final_grades = []
+        final_grades = mmtrack.get_all_final_grades()
+        for course_id, final_grade in final_grades.items():
+            course_run = CourseRun.objects.get(edx_course_key=course_id)
```

Comment:
Can you also change this to use something like `CourseRun.objects.filter(edx_course_key__in=course_ids)`?